x-pack/plugins/transform/public/app/sections/create_transform/components/step_define/common/percentiles_agg/config.ts

```diff
@@ -40,8 +40,36 @@ function parsePercentsInput(inputValue: string | undefined) {
 }
 
 // Input string should only include comma separated numbers
-function isValidPercentsInput(inputValue: string) {
-  return /^[0-9]+(,[0-9]+)*$/.test(inputValue);
+function validatePercentsInput(inputValue: unknown): {
+  isValid: boolean;
+  errorType?: ValidationResultErrorType;
+} {
+  if (typeof inputValue !== 'string') {
+    return { isValid: false, errorType: 'INVALID_FORMAT' };
+  }
+
+  // Remove all whitespace
+  const testInputValue = inputValue.split(' ').join('');
+
+  // Explicitly check for -0 to keep consistency with negative numbers
+  if (testInputValue.includes('-0')) {
+    return { isValid: false, errorType: 'NEGATIVE_NUMBER' };
+  }
+
+  // Check if it is a valid comma-separated list of numbers (including negative numbers)
+  if (/^-?\d+(?:,-?\d+)*$/.test(testInputValue)) {
+    const numbers = testInputValue.split(',').map(Number);
+    // Check for negative numbers
+    if (numbers.some((num) => num < 0)) {
+      return { isValid: false, errorType: 'NEGATIVE_NUMBER' };
+    }
+    // Check for numbers greater than 100
```

Comment:
I added this check because I'm not sure whether we should accept percentile values higher than `100`